pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -178,3 +179,150 @@ class Meta:
         advisory = UpdateRecord(**cr_update)
         # advisory.digest = digest
         return advisory, relations
+
+
+class Pulp2Modulemd(Pulp2to3Content):
+    """
+    Pulp 2to3 detail content model to store Pulp2 Modulemd content details.
+    """
+
+    # Unit key fields
+    name = models.TextField()
+    stream = models.TextField()
+    version = models.BigIntegerField()
+    context = models.TextField()
+    arch = models.TextField()
+
+    artifacts = JSONField()
+    checksum = models.TextField()
+    dependencies = JSONField()
+
+    pulp2_type = 'modulemd'
+
+    class Meta:
+        unique_together = (
+            'name', 'stream', 'version', 'context', 'arch', 'pulp2content')
+        default_related_name = 'modulemd_detail_model'
+
+    @property
+    def expected_digests(self):
+        """Return expected digests."""
+        return {'sha256': self.checksum}
+
+    @property
+    def expected_size(self):
+        """Return expected size."""
+        return
+
+    @property
+    def relative_path_for_content_artifact(self):
+        """Return relative path."""
+        relative_path = '{}{}{}{}{}snippet'.format(
+                        self.name, self.stream, self.version,
+                        self.context, self.arch)
+
+        return relative_path
+
+    @classmethod
+    async def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 modulemd content with all the fields needed to create a Pulp 3 content.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+
+        """
+        pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
+        pulp2_ids = pulp2_id_obj_map.keys()
+        pulp2_content_batch = pulp2_models.Modulemd.objects.filter(id__in=pulp2_ids)
+        pulp2modules_to_save = [
+            cls(name=md.name,
+                stream=md.stream,
+                version=md.version,
+                context=md.context,
+                arch=md.arch,
+                dependencies=md.dependencies,
+                artifacts=md.artifacts,
+                checksum=md.checksum,
+                pulp2content=pulp2_id_obj_map[md.id])
+            for md in pulp2_content_batch]
+        cls.objects.bulk_create(pulp2modules_to_save, ignore_conflicts=True)
+
+    async def create_pulp3_content(self):
+        """
+        Create a Pulp 3 Module content for saving it later in a bulk operation.
+        """
+        return Modulemd(name=self.name, stream=self.stream, version=self.version,
+                        context=self.context, arch=self.arch, artifacts=self.artifacts,
+                        dependencies=self.dependencies)
```

Comment:
Do we store dependencies in pulp3 in exactly the same format as in pulp2?

Reply:
migrated unit:
```
            "dependencies": [
                {
                    "platform": [
                        "f28"
                    ],
                    "walrus": [
                        "-5.21"
                    ]
                }
            ],
```
sycned unit directly  in pulp3:
```
            "dependencies": [
                {
                    "platform": [
                        "f28"
                    ],
                    "walrus": [
                        "-5.21"
                    ]
                }
            ],
```